contracts/src/fp/OPSuccinctEntryPoint.sol

```diff
@@ -0,0 +1,196 @@
+     */
+    function addCredit(address _user, uint256 _amount) external payable {
+        credit[_user] += _amount;
+        emit CreditAdded(_user, _amount);
+    }
+
+    /**
+     * @notice Allows a user to claim all of their credits accumulated.
+     */
+    function claimCredit(address _recipient) external {
+        uint256 creditToClaim = credit[_recipient];
+        credit[_recipient] = 0;
+
+        if (creditToClaim == 0) revert NoCreditToClaim();
+
+        (bool success,) = _recipient.call{value: creditToClaim}(hex"");
+        if (!success) revert BondTransferFailed();
+    }
+
+    /**
+     * @notice Creates a new `OPSuccinctFaultDisputeGame` via the `disputeGameFactory`, passing `_rootClaim` and `_extraData`.
+     * @param _rootClaim The root claim to initialize the new game with.
+     * @param _extraData The extra data to initialize the new game with.
+     * @dev Only whitelisted proposers are allowed to call this function.
+     * @dev The extra data includes the l2BlockNumber and the parentIndex.
+     *      Address of the entry point is appended to the extra data to ensure
+     *      the game is created through the entry point.
+     */
+    function createGame(Claim _rootClaim, bytes calldata _extraData)
+        external
+        payable
+        onlyProposer
+        returns (address newGameAddress)
+    {
+        // Append address(msg.sender) to the extraData
+        bytes memory extraDataWithClaimant = abi.encodePacked(_extraData, msg.sender);
+
+        // Call the factory to create the game
+        // Append address(msg.sender) to the extraData to record the claimant
+        IDisputeGame newGame = disputeGameFactory.create{value: msg.value}(gameType, _rootClaim, extraDataWithClaimant);
+
+        // Emit an event with the new game address and the rootClaim
+        emit CreatedOPSuccinctFaultDisputeGame(newGameAddress = address(newGame), msg.sender, _rootClaim);
+    }
+
+    /**
+     * @notice Challenges an `OPSuccinctFaultDisputeGame`.
+     * @dev Only whitelisted challengers can call this function.
+     * @dev Exact amount of ETH for proof reward is required when challenging.
+     */
+    function challengeGame(IDisputeGame _game) external payable onlyChallenger {
+        OPSuccinctFaultDisputeGame(address(_game)).challenge{value: msg.value}(msg.sender);
+    }
+
+    /**
+     * @notice Proves an `OPSuccinctFaultDisputeGame`.
+     * @dev Anyone can be a prover.
+     */
+    function proveGame(IDisputeGame _game, bytes calldata proofBytes) external {
+        OPSuccinctFaultDisputeGame(address(_game)).prove(msg.sender, proofBytes);
+    }
+
+    /**
+     * @notice Resolves an `OPSuccinctFaultDisputeGame`.
+     * @dev Anyone can resolve a game.
+     */
+    function resolveGame(IDisputeGame _game) external {
+        OPSuccinctFaultDisputeGame(address(_game)).resolve();
+    }
+
+    /// @notice Allows the owner to recover by pulling ETHs out of the contract.
```

Comment:
You should add another method which allows reclaiming ETH from a specific user that's `onlyOwner` gated, mirroring the logic inside of the OP FP contracts.